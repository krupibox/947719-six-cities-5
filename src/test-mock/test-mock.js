export const TestMock = {

  cities: [`Paris`, `Brussels`, `Hamburg`, `Amsterdam`, `Cologne`],

  review: {
    user: {
      id: 1,
      avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg`,
      name: `Max`,
    },
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    date: `2020-11-05T09:55:08.500Z`,
    rating: 4.1,
  },

  reviews: [{
    user: {
      id: 1,
      avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg`,
      name: `Max`,
    },
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    date: `2020-11-05T09:55:08.500Z`,
    rating: 4.1,
  }],

  favorites: {
    Cologne: [{
      id: 1,
      city: {
        name: `Paris`,
        location: {
          latitude: 0,
          longitude: 0,
        }
      },
      previewImage: `img/apartment-01.jpg`,
      images: [
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-03.jpg`,
      ],
      type: `Apartment`,
      price: 80,
      rating: 3.1,
      title: `Beautiful &amp; luxurious apartment at great location`,
      coordinates: [52.3909553943508, 4.85309666406198],
      host: {
        id: 1,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: `Discover daily local life in city center.`,
      isPremium: true,
      location: {
        latitude: 0,
        longitude: 0,
      }
    }],
    Hamburg: [{
      id: 1,
      city: {
        name: `Paris`,
        location: {
          latitude: 0,
          longitude: 0,
        }
      },
      previewImage: `img/apartment-01.jpg`,
      images: [
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-03.jpg`,
      ],
      type: `Apartment`,
      price: 80,
      rating: 3.1,
      title: `Beautiful &amp; luxurious apartment at great location`,
      coordinates: [52.3909553943508, 4.85309666406198],
      host: {
        id: 1,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: `Discover daily local life in city center.`,
      isPremium: true,
      location: {
        latitude: 0,
        longitude: 0,
      }
    }],
  },

  offer: {
    id: 1,
    city: {
      name: `Paris`,
      location: {
        latitude: 0,
        longitude: 0,
      }
    },
    previewImage: `img/apartment-01.jpg`,
    images: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
    ],
    type: `Apartment`,
    price: 80,
    rating: 3.1,
    title: `Beautiful &amp; luxurious apartment at great location`,
    coordinates: [52.3909553943508, 4.85309666406198],
    host: {
      id: 1,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Discover daily local life in city center.`,
    isPremium: true,
    isFavorite: true,
    location: {
      latitude: 0,
      longitude: 0,
    }
  },

  offers: [
    {
      id: 1,
      city: {
        name: `Paris`,
        location: {
          latitude: 0,
          longitude: 0,
        }
      },
      previewImage: `img/apartment-01.jpg`,
      images: [
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-03.jpg`,
      ],
      type: `Apartment`,
      price: 80,
      rating: 3.1,
      title: `Beautiful &amp; luxurious apartment at great location`,
      coordinates: [52.3909553943508, 4.85309666406198],
      host: {
        id: 1,
        name: `Angelina`,
        isPro: true,
        avatarUrl: `img/avatar-angelina.jpg`
      },
      description: `Discover daily local life in city center.`,
      isPremium: true,
      isFavorite: true,
      location: {
        latitude: 0,
        longitude: 0,
      }
    }
  ],
};


// const initialState = {
//   offers: [],
//   favorites: [],
//   cities: [],
//   activeCity: ``,
//   activeOfferId: ``,
//   activeCoords: {}
// };
