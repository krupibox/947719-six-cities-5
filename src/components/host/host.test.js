import Host from './host';
import {TestMock} from '../../test-mock/test-mock';

describe(`<Host/>`, () => {
  it(`should render correctly`, () => {
    const tree = renderer
            .create(
                <Host
                  description={TestMock.offer.description}
                  {...TestMock.offer.host}
                />)
            .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
