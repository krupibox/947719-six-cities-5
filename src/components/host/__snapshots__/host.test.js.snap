// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Host/> should render correctly 1`] = `
<div
  className="property__host"
>
  <h2
    className="property__host-title"
  >
    Meet the host
  </h2>
  <div
    className="property__host-user user"
  >
    <div
      className="property__avatar-wrapper undefined user__avatar-wrapper"
    >
      <img
        alt="Host avatar"
        className="property__avatar user__avatar"
        height={74}
        width={74}
      />
    </div>
    <span
      className="property__user-name"
    >
      Angelina
    </span>
  </div>
  <div
    className="property__description"
  >
    <p
      className="property__text"
    >
      Discover daily local life in city center.
    </p>
  </div>
</div>
`;
